drivers/sensor/max17262/max17262.c

```diff
@@ -0,0 +1,342 @@
+ * @return 0 for success
+ * @return -EINVAL if the I2C controller could not be found
+ */
+static int max17262_gauge_init(const struct device *dev)
+{
+	struct max17262_data *data = dev->data;
+	const struct max17262_config *const config = dev->config;
+	int16_t tmp, hibcfg;
+
+	data->i2c = device_get_binding(config->i2c_bus);
+	if (!data->i2c) {
+		LOG_ERR("Could not get pointer to %s device", config->i2c_bus);
+		return -EINVAL;
+	}
+
+	/* Read Status register */
+	max17262_reg_read(dev, STATUS, &tmp);
+
+	if (!(tmp & STATUS_POR)) {
+		/*
+		 * Status.POR bit is set to 1 when MAX17262 detects that
+		 * a software or hardware POR event has occurred and
+		 * therefore a custom configuration needs to be set...
+		 * If POR event did not happen (Status.POR == 0), skip
+		 * init and continue with measurements.
+		 */
+		LOG_DBG("No POR event detected - skip device configuration");
+		return 0;
+	}
+	LOG_DBG("POR detected, setting custom device configuration...");
+
+	/** STEP 1 */
+	max17262_reg_read(dev, FSTAT, &tmp);
+
+	/* Do not continue until FSTAT.DNR bit is cleared */
+	while (tmp & FSTAT_DNR)
+		max17262_reg_read(dev, FSTAT, &tmp);
+
+	/** STEP 2 */
+	/* Store original HibCFG value */
+	max17262_reg_read(dev, HIBCFG, &hibcfg);
+
+	/* Exit Hibernate Mode step 1 */
+	max17262_reg_write(dev, SOFT_WAKEUP, 0x0090);
+	/* Exit Hibernate Mode step 2 */
+	max17262_reg_write(dev, HIBCFG, 0x0000);
+	/* Exit Hibernate Mode step 3 */
+	max17262_reg_write(dev, SOFT_WAKEUP, 0x0000);
+
+	/** STEP 2.1 --> OPTION 1 EZ Config (No INI file is needed) */
+	/* Write DesignCap */
+	max17262_reg_write(dev, DESIGN_CAP, config->design_cap);
+
+	/* Write IChgTerm */
+	max17262_reg_write(dev, ICHG_TERM, config->desired_charging_current);
+
+	/* Write VEmpty */
+	max17262_reg_write(dev, VEMPTY, ((config->empty_voltage / 10) << 7) |
+					  ((config->recovery_voltage / 40) & 0x7F));
+
+	/* Write ModelCFG */
+	if (config->charge_voltage > 4275)
+		max17262_reg_write(dev, MODELCFG, 0x8400);
+	else
+		max17262_reg_write(dev, MODELCFG, 0x8000);
+
+	/*
+	 * Read ModelCFG.Refresh (highest bit),
+	 * proceed to Step 3 when ModelCFG.Refresh == 0
+	 */
+	max17262_reg_read(dev, MODELCFG, &tmp);
+
+	/* Do not continue until ModelCFG.Refresh == 0 */
+	while (tmp & MODELCFG_REFRESH)
+		max17262_reg_read(dev, MODELCFG, &tmp);
+
+	/* Restore Original HibCFG value */
+	max17262_reg_write(dev, HIBCFG, hibcfg);
+
+	/** STEP 3 */
+	/* Read Status register */
+	max17262_reg_read(dev, STATUS, &tmp);
+
+	/* Clear PowerOnReset bit */
+	tmp &= ~STATUS_POR;
+	max17262_reg_write(dev, STATUS, tmp);
+
+	return 0;
+}
+
+static const struct sensor_driver_api max17262_battery_driver_api = {
+	.sample_fetch = max17262_sample_fetch,
+	.channel_get = max17262_channel_get,
+};
+
+#define MAX17262_INIT(n)						\
+	static struct max17262_data max17262_driver_##n;		\
+									\
+	static const struct max17262_config max17262_config_##n = { \
+		.i2c_bus = DT_INST_BUS_LABEL(n),			\
```

Comment:
```suggestion
		.i2c = DEVICE_DT_GET(DT_BUS(DT_DRV_INST(n))),			\
```

You can use `DT_INST_BUS(n)` instead if https://github.com/zephyrproject-rtos/zephyr/pull/30536 is merged before this PR.

Reply:
Implemented.